src/FakeItEasy/Core/TypeCatalogueInstanceProvider.cs

```diff
@@ -32,7 +32,7 @@ public IEnumerable<T> InstantiateAllOfType<T>()
             {
                 try
                 {
-                    result.Add((T)Activator.CreateInstance(type));
+                    result.Add((T)Activator.CreateInstance(type)!);

```

Comment:
Activator.CreateInstance can only return null if the type argument is `Nullable<T>`.